Changes for production

spacemansteve opened the pull request:
Mostly changes to meta tags, also support to process a file of bibcodes.

Commit:
tweaks towards production

improved log messages, eliminated one check on returned web page, added support for file of bibcodes

Commit:
template requires prerendered

or the page does not properly load and rehydrate

Commit:
more meta tag tweaks

romanchyla commented on adstb/app.py:
```diff
@@ -116,13 +116,18 @@ def _update_timestamps(self, message):
 
     def _load_url(self, url):
         try:
+            self.logger.info('trying to load url {} at endpoint: {}'.format(url, self.conf.get('PUPPETEER_ENDPOINT')))
```
please turn them to `logger.debug` calls

romanchyla commented on adstb/app.py:
```diff
@@ -294,45 +301,38 @@ def _retrieve_abstract_template(self, url):
-            end += 1
-        
-            
+        end = html.find('</head')
```
this is likely unsafe; you may be removing too much stuff (if you remove something important, bbb will be broken)

spacemansteve replied:
I don't think I added this, perhaps we should talk about what you'd like instead.

romanchyla replied:
i think you did, i'm looking at your changes ;-)

my comment is about line 318; if we search for data-highwire, we get the beginning of the tags section -- but there can be lots of stuff between that and the end of the `</head>`

so you are potentially replacing too much; was there a problem with the original lines 317-322?

spacemansteve replied:
thx, reverted code back to original

romanchyla commented on run.py:
```diff
@@ -85,6 +83,8 @@ def submit(bibcode):
             app.logger.info('Done: {0}/{1}'.format(start, j['response']['numFound']))
             print 'Done submitting: {0}/{1}'.format(start, j['response']['numFound'])
         except Exception, e:
+            import pdb
```
to be removed

romanchyla commented on run.py:
```diff
@@ -119,18 +117,22 @@ def harvest_by_null(queue='priority-bumblebee',
-        r = client.get(url, params=params)
+        # reuse the http ads api client with keep-alive connections and bearer token
+        r = app._client.get(url, params=params)
+        import pdb
```
dtto

romanchyla commented on run.py:
```diff
@@ -119,18 +117,22 @@ def harvest_by_null(queue='priority-bumblebee',
             
-            tasks.task_priority_queue.delay(msg)
+                app.logger.info('harvest_by_null queuing target {}'.format(d['target']))
+                tasks.task_priority_queue.delay(msg)
```
here i'd recommend using logging to warn the user something, that was explicitly requested, was rejected and won't be processed

romanchyla commented on run.py:
```diff
@@ -158,6 +160,17 @@ def harvest_by_null(queue='priority-bumblebee',
     print i, last_id
 
+    
+def iscachable(target):
```
if you make this part of the application (method), then various calls can provide/override this filter - just a thought

romanchyla reviewed (commented):
please check the unittests and see some comments below, at least one of them - the bbb template `</head>` is important

Commit:
address PR issues and some pep-8 cleanup

romanchyla commented:
please fix the failing unittests @spacemansteve

Commit:
fixes for title and no javascript

to support page load when javascript is disabled we must also delete noscript with style that hides the abstract

Commit:
improve removing style tag

fix typo, make check for style more flexible, add unit test

romanchyla commented on adstb/app.py:
```diff
@@ -341,13 +334,26 @@ def _retrieve_abstract_template(self, url):
-        
-        if 'bibcode' in parts and parts['bibcode']:
-            html = html.replace(parts['bibcode'], u'{{bibcode}}')
-        
```
this should be kept (and used if it's not already); otherwise links will be wrong

spacemansteve replied:
updated pr

romanchyla commented on adstb/app.py:
```diff
@@ -341,13 +334,26 @@ def _retrieve_abstract_template(self, url):
-        
+
+        # finally, cut out noscript warning that says javascript is required
+        x = html.find('<div id="noscriptmsg"')  # div id within noscript
```
you may want to search for `id="noscriptmsg"` - that's guaranteed to be formatted that way

spacemansteve replied:
yep, changed

romanchyla commented on adstb/bumblebee.py:
```diff
@@ -212,8 +231,39 @@ def build_meta_tags(solrdoc):
+        data['title'] = _html_to_text(data['title'][0])
+    link_url_template = 'http://dev.adsabs.harvard.edu/abs/{}'
+    if 'LINK_URL_TEMPLATE' in conf:
+        data['url'] = conf['LINK_URL_TEMPLATE'].format(data['bibcode'])
```
i'm confused by this variable, it is not used in the tags sections; and also, i think it shouldn't be, the template has been generated maybe from a different domain that what's in the config

(which brings up the sensitive point of "should we verify/sanitize links in the template" ?)

spacemansteve replied:
updated pr

romanchyla reviewed (changes requested):
please see comments

Commit:
remove less of html header

Commit:
changed search for div with id=noscriptmsg to be more flexible

Commit:
replace bibcodes in page

Commit:
removed link url

romanchyla commented on adstb/bumblebee.py:
```diff
@@ -212,8 +231,34 @@ def build_meta_tags(solrdoc):
+def _html_to_text(t):
+    """remove html tags and clean html encoded characters and return plain text string"""
+    t = htmlParser.unescape(t)
+    t = tag_re.sub('', t)
```
shouldn't the lines 257-258 be swapped?

spacemansteve replied:
I like what it currently does, converting u'Renormalization of quark propagators from twisted-mass lattice QCD at N&lt;SUB&gt;f&l\
t;/SUB&gt;=2' to u'Renormalization of quark propagators from twisted-mass lattice QCD at Nf=2'.  What did you have in mind?

romanchyla reviewed (approved):
Just a small comment below

Also, can you please check the failing travis? I do realize that the tests are reported as failing for a long time (maybe they never went green?) but it is likely something silly, they should work before me merge.

I'm not seeing any major change in the code; are we satisfied that the current code generates proper static pages for turbobee?

romanchyla commented:
you sure about this latest change? the template will miss {{bibcode}}
placeholders...

On Mon, Jun 17, 2019 at 1:32 PM Steve McDonald <notifications@github.com>
wrote:

> *@spacemansteve* commented on this pull request.
> ------------------------------
>
> In adstb/app.py
> <https://github.com/adsabs/ADSTurboBeePipeline/pull/2#discussion_r294408560>
> :
>
> >          html = html[0:x] + '{{abstract}}' + html[end:]
> -
> -        if 'bibcode' in parts and parts['bibcode']:
> -            html = html.replace(parts['bibcode'], u'{{bibcode}}')
> -
>
> updated pr
>
> —
> You are receiving this because your review was requested.
> Reply to this email directly, view it on GitHub
> <https://github.com/adsabs/ADSTurboBeePipeline/pull/2?email_source=notifications&email_token=AADERESH4WLGC2VY6QCUTETP27DELA5CNFSM4HRMB6JKYY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOB3YGYCY#discussion_r294408560>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AADERERLF6UWT4PWGZAW4N3P27DELANCNFSM4HRMB6JA>
> .
>

Commit:
changes for unit tests in CI

Commit:
testing CI

Commit:
config.py should default to sqlite

spacemansteve commented:
Checks pass on github after a change to config.py
I'm not sure about "you sure about this latest change? the template will miss {{bibcode}}
placeholders...".  I'll thought you wanted this included.  I will verify after PR has been merged.
